### PR TITLE
[FS-200]: Delete Family Api Integration

### PR DESCRIPTION
**Pull request by erfan-goodarzi**
## Changes

- Create `DeleteFamilyMutation`.
- Integrate the mutation hook with family buttons. 
- Add failure and success feedback.

### CODE REVIEW

**Comment by erfan-goodarzi** [comment]
@ASafaeirad @AmirabbasJ 
Guys, we have duplication in the `FamilyDetail` and `FamilyTableRow` files. what do u think? what's ur suggestions?

**Comment by ASafaeirad** [comment]
> @ASafaeirad @AmirabbasJ Guys, we have duplication in the `FamilyDetail` and `FamilyTableRow` files. what do u think? what's ur suggestions?

What part is duplicated?

File: app/pages/Dashboard/Families/FamilyDetail/FamilyDetail.tsx [hunk 1]
```diff
@@ -51,7 +55,38 @@ export const FamilyDetail = () => {
             leftIcon={<TrashIcon width="18" height="18" />}
             px="lg"
             py="8px"
-            onClick={() => openDeleteFamilyModal(family.name)}
+            onClick={() =>
+              openDeleteFamilyModal({
+                name: family.name,
+                onDeleteFamily: async () => {
+                  // FIXME: maybe abstract
+                  try {
```
Comment (erfan-goodarzi) [inline app/pages/Dashboard/Families/FamilyDetail/FamilyDetail.tsx:63]: @ASafaeirad here.

File: app/components/FamilyList/FamilyTableRow.tsx [hunk 2]
```diff
@@ -34,7 +41,33 @@ export const FamilyTableRow = ({ order, family }: Props) => {
           <FamilyActionButton
             onDelete={e => {
               e.stopPropagation();
-              openDeleteFamilyModal(name);
+              openDeleteFamilyModal({
+                name,
+                onDeleteFamily: async () => {
+                  // FIXME: maybe abstract
+                  try {
```
Comment (erfan-goodarzi) [inline app/components/FamilyList/FamilyTableRow.tsx:48]: @ASafaeirad And here.

Comment (erfan-goodarzi) [inline app/components/FamilyList/FamilyTableRow.tsx:48]: @ASafaeirad And here.

File: app/components/DeleteFamilyModal/DeleteFamilyModal.tsx [hunk 3]
```diff
@@ -5,12 +5,21 @@ import { deleteFamilyModalIds as Ids } from './DeleteFamilyModal.ids';
 
 const t = messages.families.list.delete.modal;
 
-export const openDeleteFamilyModal = (name: string) =>
+interface DeleteFamilyModalProps {
```
Comment (ASafaeirad) [inline app/components/DeleteFamilyModal/DeleteFamilyModal.tsx:8]:
```suggestion
interface Props {
```

File: app/components/DeleteFamilyModal/DeleteFamilyModal.tsx [hunk 4]
```diff
@@ -5,12 +5,21 @@ import { deleteFamilyModalIds as Ids } from './DeleteFamilyModal.ids';
...
+  name: string;
+  onDeleteFamily: () => void;
+}
+
+export const openDeleteFamilyModal = ({
+  name,
+  onDeleteFamily,
+}: DeleteFamilyModalProps) =>
```
Comment (ASafaeirad) [inline app/components/DeleteFamilyModal/DeleteFamilyModal.tsx:16]:
```suggestion
}: Props) =>
```

File: app/components/DeleteFamilyModal/DeleteFamilyModal.tsx [hunk 5]
```diff
@@ -5,12 +5,21 @@ import { deleteFamilyModalIds as Ids } from './DeleteFamilyModal.ids';
 
 const t = messages.families.list.delete.modal;
 
-export const openDeleteFamilyModal = (name: string) =>
+interface DeleteFamilyModalProps {
+  name: string;
+  onDeleteFamily: () => void;
```
Comment (ASafaeirad) [inline app/components/DeleteFamilyModal/DeleteFamilyModal.tsx:10]:
```suggestion
  onDeleteFamily: VoidFunction;
```

File: app/components/FamilyList/FamilyTableRow.tsx [hunk 6]
```diff
@@ -34,7 +41,33 @@ export const FamilyTableRow = ({ order, family }: Props) => {
...
+                    const { data } = await DeleteFamilyMutation({
+                      variables: { id },
+                    });
+
+                    if (isNull(data)) throw new Error('data is null');
+                    showNotification({
+                      title: t.notification.title,
+                      message: t.notification.successfulDeleted(
```
Comment (ASafaeirad) [inline app/components/FamilyList/FamilyTableRow.tsx:56]: `t.notification.delete.success` is a better name

File: app/components/FamilyList/FamilyTableRow.tsx [hunk 7]
```diff
@@ -34,7 +41,33 @@ export const FamilyTableRow = ({ order, family }: Props) => {
...
+                        data.family.name,
+                      ),
+                      type: 'success',
+                    });
+                  } catch (err) {
+                    console.error('error occurred', err);
+                    showNotification({
+                      title: t.notification.title,
+                      message: t.notification.failedDeleted(name),
```
Comment (ASafaeirad) [inline app/components/FamilyList/FamilyTableRow.tsx:65]: `t.notification.delete.failed` is a better name

File: app/messages.ts [hunk 8]
```diff
@@ -76,6 +76,13 @@ export const messages = {
       title: 'لیست خانواده ها',
       delete: {
         modal: {
+          notification: {
+            title: 'حذف خانواده',
+            successfulDeleted: (name: string) =>
+              `خانواده ی “${name}” با موفقیت حذف شد.`,
```
Comment (ASafaeirad) [inline app/messages.ts:82]: خانوار "name" با موفقیت حذف شد.

File: app/messages.ts [hunk 9]
```diff
@@ -76,6 +76,13 @@ export const messages = {
...
+            failedDeleted: (name: string) =>
+              `مشکلی در مرحله حذف خانواده ی "${name}" بوجود آمده است. لطفا دوباره تلاش کنید.`,
```
Comment (ASafaeirad) [inline app/messages.ts:84]: * خانوار

File: libs/data-layer/operations/mutations/useDeleteFamilyMutation.ts [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+interface Variables {
+  id: string;
+}
+
+const toApiVariables = (
+  variables?: Variables | null,
+): ApiDeleteFamilyMutationMutationVariables | undefined =>
+  variables == null
+    ? undefined
```
Comment (ASafaeirad) [inline libs/data-layer/operations/mutations/useDeleteFamilyMutation.ts:44]: variables should not be null. please improve the type

File: libs/data-layer/operations/mutations/useDeleteFamilyMutation.ts [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    : {
+        id: variables.id,
+      };
+
+export const useDeleteFamilyMutation = (
+  options?: MutationHookOptions<
+    ApiDeleteFamilyMutationMutation,
+    ApiDeleteFamilyMutationMutationVariables
+  >,
```
Comment (ASafaeirad) [inline libs/data-layer/operations/mutations/useDeleteFamilyMutation.ts:53]: Please update this type regarding the latest PR.
We've created a new type for mutation options.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**